MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -392,4 +399,55 @@ - (void)notifyFailureBeforeSendingForItem:(id<MSLog>)item withError:(NSError *)e
   }
 }
 
+#pragma mark – Observers
+
+- (void)addObservers {
+
+// There is no need to do trigger sending on macOS because we can just continue to execute tasks there.
+#if !TARGET_OS_OSX
+  if (!MS_IS_APP_EXTENSION) {
+    __weak typeof(self) weakSelf = self;
+    if (self.appDidEnterBackgroundObserver == nil) {
+      void (^notificationBlock)(NSNotification *note) = ^(NSNotification __unused *note) {
+        typeof(self) strongSelf = weakSelf;
+        if (self.timerSource != nil) {
+          [strongSelf flushQueue];
```

Comment:
We should also suspend the channels, this will in turn release the http session which is not configured to run in the background. This will minimize our impact on the system while in background.

Reply:
I've added a call to `suspend` in the handler that gets called when is running out of background time. I'm not sure if this is sufficient. I need to think some more.

Reply:
Yep that should do it, maybe doing some tests to make sure it suspends in a timely manner.